src/shared/components/paginationControls/PaginationControls.tsx

```diff
@@ -45,24 +49,70 @@ export class TablePaginationControls extends React.Component<ITablePaginationCon
         showItemsPerPageSelector:true,
         showFirstPage:false,
         showLastPage:false,
-        onChangeItemsPerPage: ()=>0,
-        onFirstPageClick:()=>0,
-        onPreviousPageClick: ()=>0,
-        onNextPageClick: ()=>0,
-        onLastPageClick: ()=>0,
         className: "",
         marginLeft: 5,
         previousPageDisabled:false,
-        nextPageDisabled:false
+        nextPageDisabled:false,
+        pageNumberEditable: false
     };
 
-    constructor(props:ITablePaginationControlsProps) {
+    constructor(props:IPaginationControlsProps) {
         super(props);
         this.handleChangeItemsPerPage = this.handleChangeItemsPerPage.bind(this);
+        this.handleCurrentPageSubmit = this.handleCurrentPageSubmit.bind(this);
+        this.handleChangeCurrentPage = this.handleChangeCurrentPage.bind(this);
+        this.handleMiddleButtonClick = this.handleMiddleButtonClick.bind(this);
+        this.handleOnBlur = this.handleOnBlur.bind(this);
+    }
+
+    @observable
+    private textBoxActive = false;
+    @observable
+    private _pageNumberValue?: number = this.props.currentPage as number;
+
+    get pageNumberValue() {
+
+        if (this._pageNumberValue !== undefined) {
+            return this._pageNumberValue;
+        } else {
+            return this.props.currentPage;
+        }
+    }
+
+    private jumpToPage() {
+
+        this.textBoxActive = false;
+        if (this.props.onChangeCurrentPage) {
+            this.props.onChangeCurrentPage(this.pageNumberValue as number);
+        }
+        this._pageNumberValue = undefined;
     }
 
     handleChangeItemsPerPage(evt:React.FormEvent<HTMLSelectElement>) {
-        (this.props.onChangeItemsPerPage || (()=>0))(parseInt((evt.target as HTMLSelectElement).value,10));
+
+        if (this.props.onChangeItemsPerPage) {
+            this.props.onChangeItemsPerPage(parseInt((evt.target as HTMLSelectElement).value,10));
+        }
+    }
+
+    handleCurrentPageSubmit(evt:React.KeyboardEvent<HTMLInputElement>) {
+
+        if(evt.charCode == 13){
+            this.jumpToPage();
+        }
+    }
+
+    handleChangeCurrentPage(evt:React.FormEvent<HTMLInputElement>) {
+
+        this._pageNumberValue = parseInt(evt.currentTarget.value);
```

Comment:
Andy's right.     I've been using defaultValue attr to avoid controlled components but I see it's necessary here because it needs to adjust to arrow navigation.